include/deal.II/lac/petsc_matrix_base.h

```diff
@@ -784,6 +784,43 @@ namespace PETScWrappers
     void Tvmult_add (VectorBase       &dst,
                      const VectorBase &src) const;
 
+    /**
+     * Perform the matrix-matrix multiplication <tt>C = A * B</tt>, or, if an
+     * optional vector argument is given, <tt>C = A * diag(V) * B</tt>, where
```

Comment:
can you explain how you indicate that the vector `V` is optional? Is it ignored if it is an empty vector, or how do you indicate that it should be ignored?

Reply:
I forgot to set `V` to an empty `VectorBase()` by default. I also explained that `V` will also be ignored if it is not compatible with `B`. Thanks.